tests/frontend/test_engine.py

```diff
@@ -27,61 +27,54 @@
 @pytest.fixture
 def eng(backend):
     """Engine fixture."""
-    return sf.Engine(backend)
+    return sf.LocalEngine(backend)
+
+@pytest.fixture
+def prog(backend):
+    """Program fixture."""
+    prog = sf.Program(2)
+    with prog.context as q:
+        ops.Dgate(0.5) | q[0]
+    return prog
 
 
 class TestEngine:
     """Test basic engine functionality"""
 
     def test_load_backend(self):
         """Backend can be correctly loaded via strings"""
-        eng = sf.Engine("base")
+        eng = sf.LocalEngine("base")
         assert isinstance(eng.backend, BaseBackend)
 
+    def test_bad_backend(self):
+        """Backend must be a string or a BaseBackend instance."""
+        with pytest.raises(TypeError, match='backend must be a string or a BaseBackend instance'):
+            eng = sf.LocalEngine(0)
+
 
 class TestEngineProgramInteraction:
     """Test the Engine class and its interaction with Program instances."""
 
-    def test_no_return_state(self, eng):
-        """Engine returns None when no state is requested"""
-        prog = sf.Program(2)
-
-        with prog.context as q:
-            ops.Dgate(0.34) | q[0]
-
-        res = eng.run(prog, return_state=False)
-        assert res is None
-
-    def test_history(self, eng):
+    def test_history(self, eng, prog):
         """Engine history."""
-        prog = sf.Program(2)
-
-        with prog.context as q:
-            ops.Dgate(0.5) | q[0]
-
         # no programs have been run
         assert not eng.run_progs
         eng.run(prog)
         # one program has been run
         assert len(eng.run_progs) == 1
         assert eng.run_progs[-1].source == prog
 
-    def test_reset(self, eng):
+    def test_reset(self, eng, prog):
         """Running independent programs with an engine reset in between."""
-        D = ops.Dgate(0.5)
-        p1 = sf.Program(2)
-        with p1.context as q:
-            D | q[1]
         assert not eng.run_progs
-        eng.run(p1)
+        eng.run(prog)
         assert len(eng.run_progs) == 1
 
         eng.reset()
         assert not eng.run_progs
-
         p2 = sf.Program(3)
         with p2.context as q:
-            D | q[2]
+            ops.Rgate(1.0) | q[2]
```

Comment:
any reason for this change?

Reply:
I guess I wanted to make the second program clearly different from the first one.

Reply:
sounds good